openmdao/core/system.py

```diff
@@ -3056,13 +3056,12 @@ def get_design_vars(self, recurse=True, get_sizes=True, use_prom_ivc=True):
                             else:
                                 out[name] = meta
 
-        if out:
-            for conn_in, conn_out in conns.items():
-                conn_out_attr = conn_out in abs2meta_out
-                if conn_out in out and conn_out_attr:
-                    if "allow_desvar" not in abs2meta_out[conn_out]['tags']:
-                        raise RuntimeError(f"Cannot connect the design variable '{conn_out}' to "
-                                           f"'{conn_in}' as an input.")
+        if out and 'model' in self.msginfo:
```

Comment:
instead of checking if `'model' in self.msginfo`, why not just check if `self is model`?

Reply:
Changed